2022/01/index.js

```diff
@@ -2,13 +2,44 @@ const { promises: fs } = require('fs');
 
 async function getInput() {
   const input = (await fs.readFile(`${__dirname}/input.txt`))
-    .toString();
+    .toString()
+    .trim();
+  const inventories = input
+    .split('\n\n')
+    .map(inventory => inventory.split('\n').map(calories => parseInt(calories)));
+
+  return inventories;
+}
+
+function sumArray(arr) {
+  return arr.reduce(
+    (acc, element) => acc + element,
+    0
+  );
+}
+
+async function getSumCalories() {
+  const inventories = await getInput();
+  const sumCalories = inventories.reduce(
```

Comment:
This is pretty inefficient, since you’re expanding the array each iteration. This should be .map.

Reply:
Oh yeah should've noticed since the initial value is an array :face-palm: